src/dso_api/dynamic_api/filterset.py

```diff
@@ -37,29 +49,61 @@ def get_filters(cls):
 
         # Apply camelCase to the filter names, after they've been initialized
         # using the model fields snake_case as input.
-        return {
-            snake_to_camel_case(attr_name): filter
-            for attr_name, filter in filters.items()
+        base_filters = {
+            snake_to_camel_case(attr_name): filter_class
+            for attr_name, filter_class in filters.items()
         }
 
+        return base_filters
 
-def filterset_factory(model: Type[DynamicModel]) -> Type[DynamicFilterSet]:
+
+def filterset_factory(model: Type[DynamicModel], serializer_class: Type[DynamicSerializer]) -> Type[DynamicFilterSet]:
     """Generate the filterset based on the dynamic model."""
     # See https://django-filter.readthedocs.io/en/master/guide/usage.html on how filters are used.
     # Determine which fields are included:
     # Excluding geometry fields for now, as the default filter only performs exact matches.
     # This isn't useful for polygon fields, and excluding it avoids support issues later.
-    fields = {f.attname: _get_field_lookups(f) for f in model._meta.get_fields()}
+    fields = {f.attname: _get_field_lookups(f.__class__) for f in model._meta.get_fields()}
+
+    filters = {}
+    camel_case_fields = list(map(snake_to_camel_case, fields))
+    for field_name in serializer_class.Meta.fields:
+        if field_name not in ["_links", "schema"] + camel_case_fields:
+            try:
+                field_schema = model._table_schema['schema']['properties'][field_name]
+            except KeyError:
+                pass
```

Comment:
```suggestion
                continue
```

en dan kan de `else:` eronder ook weg